src/core/symbology/qgssymbol.h

```diff
@@ -867,6 +868,23 @@ class CORE_EXPORT QgsSymbol
      */
     void stopFeatureRender( const QgsFeature &feature, QgsRenderContext &context, int layer = -1 );
 
+    /**
+     * Returns the symbol's extent buffer.
+     *
+     * \returns The symbol's extent buffer
+     * \since QGIS 3.42
+     */
+    double extentBuffer() const;
+
+    /**
+     * Sets the symbol's extent buffer.
+     *
```

Comment:
This needs some comment about permitting a negative buffer size, which would cause the symbol not to be rendered if it's close to the edge of the map.

But that said -- negative sizes won't work well for anything but single symbol renderer, because if you have any symbol in the other renderers with the default 0 size then maximumExtentBuffer will ignore the negative values and just return 0...

Are negative buffers necessary? I'd drop them if not